numpy/core/include/numpy/ndarraytypes.h

```diff
@@ -159,9 +159,10 @@ enum NPY_TYPECHAR {
 typedef enum {
         NPY_QUICKSORT=0,
         NPY_HEAPSORT=1,
-        NPY_MERGESORT=2
+        NPY_MERGESORT=2,
+        NPY_STABLESORT=3
 } NPY_SORTKIND;
-#define NPY_NSORTS (NPY_MERGESORT + 1)
+#define NPY_NSORTS (NPY_STABLESORT + 1)
```

Comment:
Is there a reason `NPY_NSORTS` is a `#define` and not just the last enum member?

Reply:
Enums start at zero.

Reply:
I'm not sure I see your point. Does this not set `NPY_NSORTS` to 4?
```
typedef enum {
        NPY_QUICKSORT=0,
        NPY_HEAPSORT=1,
        NPY_MERGESORT=2,
        NPY_STABLESORT=3,
        NPY_NSORTS
} NPY_SORTKIND;

Reply:
Yes, that would work. The current method was inherited and is probably > 13+ years old at this point.

Reply:
Should I make this change?

Although this would work and I see the advantage of not having to keep N_SORTS up to date manually, I feel it a little unsettling because N_SORTS does not logically belong to that place. The enum stores different kinds of sorts and 0, 1, 2, 3 are there just because the C language works in that way.

Reply:
no, I'd leave it. having it separate increases type safety a bit as the compiler can warn about invalid entries.